### PR TITLE
Adds a test for validate_config

### PR DESCRIPTION
**Pull request by oyeb**
* This test checks if an `ArgumentError` exception is raised when some config is missing.
* Also removes an unused variable from `application.ex`.

### CODE REVIEW

File: test/gringotts_test.exs [hunk 1]
```diff
@@ -1,11 +1,17 @@
 defmodule GringottsTest do
   use ExUnit.Case
 
-  alias Gringotts.Worker
   import Gringotts
 
+  @test_config [adapter: GringottsTest.FakeGateway,
+                some_auth_info: :merchant_secret_key,
```
Comment (pkrawat1) [inline test/gringotts_test.exs]: @oyeb correct this formatting.

File: test/gringotts_test.exs [hunk 2]
```diff
@@ -36,10 +42,8 @@ defmodule GringottsTest do
     end
   end
 
-  setup_all do
-    Application.put_env(:gringotts, GringottsTest.FakeGateway, [
-          adapter: GringottsTest.FakeGateway,
-          some_auth_info: :merchant_secret_key])
+  setup do
+    Application.put_env(:gringotts, GringottsTest.FakeGateway, @test_config)
     :ok
```
Comment (pkrawat1) [inline test/gringotts_test.exs]: why this `:ok`?

File: lib/gringotts/application.ex [hunk 3]
```diff
@@ -8,8 +8,8 @@ defmodule Gringotts.Application do
   def start(_type, _args) do
     import Supervisor.Spec, warn: false
     app_config = Application.get_all_env(:gringotts)
-    adapters = Enum.filter(app_config, fn({key, klist}) -> klist != [] end)
-                |> Enum.map(fn({key, klist}) -> Keyword.get(klist, :adapter) end)
+    adapters = Enum.filter(app_config, fn({_, klist}) -> klist != [] end)
+                |> Enum.map(fn({_, klist}) -> Keyword.get(klist, :adapter) end)
```
Comment (pkrawat1) [inline lib/gringotts/application.ex]: @oyeb thanks for this!

File: test/gringotts_test.exs (hunk 2 above)
Comment (oyeb) [reply test/gringotts_test.exs]: `setup` and `setup_all` macros will merge the block's return value into the test contexts, if nothing is to be inserted into context, an `:ok` must be returned. [ExUnit docs](https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#module-context)

**Pull request merged**